src/test/kotlin/no/skatteetaten/aurora/boober/feature/ToxiproxySidecarFeatureTest.kt

```diff
@@ -304,7 +365,279 @@ class ToxiproxySidecarFeatureTest : AbstractMultiFeatureTest() {
+                            "enabled": true,
+                            "proxyname": "proxy1"
+                        },
+                        "db2": {
+                            "enabled": true,
+                            "proxyname": "proxy2"
+                        }
+                    }
+                },
+                "database": {
+                    "db1": {"enabled": true},
+                    "db2": {"enabled": true}
+                }
+            }""",
+            mutableSetOf(createEmptyService(), createEmptyDeploymentConfig()),
+            createdResources = 3
+        )
+
+        assertThat(serviceResource)
+            .auroraResourceModifiedByThisFeatureWithComment(
+                comment = "Changed targetPort to point to toxiproxy",
+                featureIndex = 2
+            )
+
+        assertThat(dcResource)
+            .auroraResourceModifiedByThisFeatureWithComment(
+                comment = "Added toxiproxy volume and sidecar container",
+                sourceIndex = 2,
+                featureIndex = 2
+            )
+
+        assertThat(secretResource1)
+            .auroraResourceModifiedByThisFeatureWithComment(
+                comment = "Changed JDBC URL to point to Toxiproxy",
+                featureIndex = 2
+            )
+
+        assertThat(secretResource2)
+            .auroraResourceModifiedByThisFeatureWithComment(
+                comment = "Changed JDBC URL to point to Toxiproxy",
+                featureIndex = 2
+            )
+
+        val jdbcUrl1 = Base64
+            .decodeBase64((secretResource1.resource as Secret).data["jdbcurl"])
+            .toString(Charset.defaultCharset())
+
+        val jdbcUrl2 = Base64
+            .decodeBase64((secretResource2.resource as Secret).data["jdbcurl"])
+            .toString(Charset.defaultCharset())
+
+        assertThat(jdbcUrl1).isEqualTo("jdbc:oracle:thin:@localhost:18000/db1")
+        assertThat(jdbcUrl2).isEqualTo("jdbc:oracle:thin:@localhost:18001/db2")
+
+        assertThat(configResource)
+            .auroraResourceCreatedByThisFeature()
+            .auroraResourceMatchesFile("configWithNamedDatabaseProxiesMapping.json")
+    }
+
+    @Test
+    fun `Should fail with an error message when there are missing databases`() {
+
+        val errorMessage = assertThrows<MultiApplicationValidationException> {
+            generateResources(
+                """{
+                    "toxiproxy": {
+                        "version": "2.1.3",
+                        "database": {"db1": true}
+                    }
+                }""",
+                createEmptyService(),
+                createEmptyDeploymentConfig()
+            )
+        }.errors.first().errors.first().message
+
+        val expectedErrorMessage =
+            "Found Toxiproxy config for database named db1, but there is no such database configured."
 
         assertThat(errorMessage).isEqualTo(expectedErrorMessage)
     }
```

Comment:
Jeg savner litt tester i forhold til forventet oppførsel med enabled: false og toxiproxy/database/<navn>: false 

Det er mange slike mutasjoner for å disable og enable. Det er greit å få noen tester på tenker jeg :) Fort å gå glipp av noe der.